web-ui/src/main/resources/catalog/components/admin/uiconfig/UiConfigDirective.js

```diff
@@ -109,13 +109,22 @@
       },
       link: function(scope, element, attrs) {
         element.val(JSON.stringify(scope.value));
-        element.on('change', function() {
-          var newValue = $(this).val();
-          try {
-            angular.merge(scope.value, JSON.parse(newValue));
-          } catch (e) {
-            console.warn('Error parsing JSON: ', newValue);
+
+        scope.$watch('value', function(newValue, oldValue, s) {
+          if (newValue !== oldValue) {
+            element.val(JSON.stringify(newValue));
           }
+        }, true);
+
+        element.on('change', function(eventObject) {
+          scope.$apply(function() {
+            var newValue = element.val();
+            try {
+              angular.merge(scope.value, JSON.parse(newValue));
```

Comment:
What is the behaviour of this on an array? Also I'm starting to think this logic is flawed because if a key/index is defined in `scope.value` but not present in `newValue` anymore it will still be there after the merge, which is counter intuitive. Agreed, this has nothing to do with your PR but was pre-existent.

Reply:
Since `angular.merge` expect objects, the array index are used as field names. For example:
```javascript
var arr = ['a', 'b', 'c'];
console.log(angular.merge(arr, ['e', 'f'])); // prints ['e', 'f', 'c']
```

Reply:
Yes, I don't know why don't assign `JSON.parse(newValue)` to `scope.value` directly. Maybe there is a case I'm not seeing.